netutils/webclient/webclient.c

```diff
@@ -903,8 +903,7 @@ int webclient_perform(FAR struct webclient_context *ctx)
       if (ret < 0)
         {
           nerr("ERROR: connect failed: %d\n", errno);
-          free(ws);
-          return ret;
+          goto errout_with_errno;
```

Comment:
Won't this end up closing the socket twice?
In case `connect` fails, line 899 already closes the socket.

Reply:
The code is changed recently, look like the leak is fixed.